library/defrag.c

```diff
@@ -371,31 +377,64 @@ void *eblob_defrag_thread(void *data)
 			continue;
 		}
 
+		pthread_mutex_lock(&b->defrag_state_lock);
 		if (b->want_defrag == EBLOB_DEFRAG_STATE_NOT_STARTED)
 			b->want_defrag = EBLOB_DEFRAG_STATE_DATA_COMPACT;
 
-		eblob_defrag(b);
+		chunks_dir = b->defrag_chunks_dir;
```

Comment:
I think I don't get an idea of such swap

Reply:
To move `defrag_chunks_dir` into local scope

Reply:
To move `defrag_chunks_dir` into local scope